crates/sui-indexer/src/models/tx_indices.rs

```diff
@@ -110,6 +119,20 @@ impl TxIndex {
         Vec<StoredTxKind>,
     ) {
         let tx_sequence_number = self.tx_sequence_number as i64;
+        let tx_affected_addresses = self
+            .recipients
+            .iter()
+            .map(|r| StoredTxAffected {
+                tx_sequence_number,
+                affected: r.to_vec(),
+                sender: self.sender.to_vec(),
+            })
+            .chain(std::iter::once(StoredTxAffected {
+                tx_sequence_number,
+                affected: self.sender.to_vec(),
+                sender: self.sender.to_vec(),
+            }))
+            .collect();
```

Comment:
It's to support filtering by sender *and* affected address. Transaction filters in GraphQL all follow the same rough shape:

- They get a table that includes the filter value mapped to transaction sequence number, and their primary key is the combination of the filter value and transaction sequence number -- this allows us to phrase a filter on just this field as a single index scan.
- They get a `sender` column, and a secondary index from the sender and the filter value to transaction sequence number to support filtering by sender and the filter value. The alternative is to join against `tx_senders` and postgres is not able to perform this join efficiently, reliably. This is why we we require that if the user sends a query that involves any other combination of filters (not sender + something) they need to provide a scan limit.
- They get a secondary index on transaction sequence number to support pruning.

By following this pattern, we are able to give people a simple rule to follow: You can apply any single filter without worrying, and you can optionally include a sender filter without worrying, but anything else requires using scan limits. Otherwise it is very difficult for users to figure out what they can do, based on the schema we advertise, very much like things are today in JSON-RPC, where the schema implies we support any number of queries that we don't, for a variety of reasons.

Reply:
Awesome thanks for the detailed information. Do we happen to have this knowledge/best practice written down anywhere in the repo?

Reply:
Good question -- the principle around using scan limits is documented in the GraphQL schema:

https://github.com/MystenLabs/sui/blob/77b18b45c195103e25d62fb6cd35ab812b9e7f12/crates/sui-graphql-rpc/src/types/query.rs#L382-L401

But the insight around the DB schema and transaction patterns is only captured in the graphql-benchmark repo:

https://github.com/MystenLabs/graphql-benchmark/blob/a7bdfad4c0d419cffa8acde24b12b373fcf7e1fa/migration_scripts/bulk_load/src/hybrid_tx.clj#L8-L49

I can add a version of this comment here:

https://github.com/MystenLabs/sui/blob/77b18b45c195103e25d62fb6cd35ab812b9e7f12/crates/sui-graphql-rpc/src/types/transaction_block/tx_lookups.rs

Reply:
Ended up splitting this out into its own PR because there were some other docs I wanted to add! #19363